src/app/form/form.tsx

```diff
@@ -0,0 +1,240 @@
+'use client';
+
+import { KeyValueList } from '@/libs/react/components/list';
+import {
+  Alert,
+  Box,
+  Button,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from '@mui/material';
+import Grid from '@mui/material/Grid2';
+import { useFormik } from 'formik';
+import { useState } from 'react';
+import * as yup from 'yup';
+
+interface IFormValues {
+  firstName: string;
+  lastName: string;
+  emailAddress: string;
+  dateOfBirth: string;
+  yearsOfExperience: number;
+  educationLevel: string;
+}
+
+enum EducationLevel {
+  HighSchool = 'high-school',
+  College = 'college',
+  Associate = 'associate-or-trade',
+  Bachelors = 'bachelors',
+  Masters = 'masters',
+  Doctorate = 'doctorate',
+}
+
+const EducationLevelDisplay = {
```

Comment:
I wonder if it's worth typing this as a Record<EducationLevel, string> rather than casting on line 100, coupled with typing line 25 as EducationLevel, should get around all the typeness here

Reply:
Had to type as `Record<string, string>` as the reason for the cast is for the index signature. But have removed the cast at least :)